### DISCUSSION TITLE
Support adding teams from other projects in Azure DevOps as reviewers/assignees to PRs

### DISCUSSION THREAD

**Discussion by markhakansson** [renovatebot/renovate#27586] [Suggest an Idea]
### Tell us more.

Currently, there is support for adding teams as reviewers and assignees to a PR that renovate creates for a repository. However, renovate only fetches the teams that belongs to the same project as said repository.

It would be a neat addition to support adding teams from other projects, as reviewers/assignees as well. As you can already do this within the platform. For example, sometimes when we want the feedback from another team in another project we usually set them as reviewers.

### Example

- `<Organization>/`
  - `<Project A>/`
     - `<Repo A>/`
  - `<Project B>/`
    - `<Repo B>/`

If renovate runs against `<Organization>/<Project A>/<Repo A>`, it currently fetches the teams that belong to `<Project A>`.
There is no support for assigning a team in `<Organization>/<Project B>` to the PR in `<Repo A>`.

**Comment by rarkins** [comment]
Does renovate need to validate that a team name is valid? Is the validation why you can't add additional teams today?

**Reply by markhakansson** [reply]
OT: Sorry for the late reply. And also great work on renovate!

Renovate does not do any extra validation really. The Azure Devops API requires the IDs of the teams in order to assign them to a PR. And these IDs are fetched from the current project that the repository to be updated belongs to. 

Which makes sense. There are extra steps required to fetch the team data from other projects. You could also allow to use the raw ID for the team or user in `reviewers: []` or `assignees: []`. But then renovate would have to be able to differentiate a name and identifier apart.

I've added an example in the original post.

**Reply by rarkins** [reply]
The example you gave seems counterintuitive. I would have expected teams from the Organization to be allowed, but not from a parallel project. Do you mean that any team from any project can be assigned to any other project within the same org?

**Reply by markhakansson** [reply]
On our organization we can add teams from parallel projects as assignees/reviewers on any PR. I do not know if this is how it works by default, or if our administrators have setup permissions to make it possible.

The more I look into it this, it seems that we may be trying to use ADO's project structure as if was similar to GitLab's groups. Instead, we should probably consolidate all projects into a single one. Which is [the recommended approach for smaller organizations](https://learn.microsoft.com/en-us/azure/devops/user-guide/plan-your-azure-devops-org-structure?view=azure-devops#how-many-projects-do-you-need).

To conclude this, it is possible to add teams from parallel projects to any projects PR in ADO. But does not seem to be a very common scenario, and adding support for that on renovate could potentially just add more complexity with very little gain as I see it.
